src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const arrayOfStates = [];
+  const stateCopy = { ...state };
+
+  for (const obj of actions) {
+    if (obj.type === 'clear') {
```

Comment:
correct, `switch` would be better fit there

Reply:
fixed